providers/google.go

```diff
@@ -124,13 +135,8 @@ func (g *GoogleOp) requestTokens(ctx context.Context, cicHash string) (*simpleoi
 	logrus.Infof("listening on http://%s/", ln.Addr().String())
 	logrus.Info("press ctrl+c to stop")
 
-	g.server = &http.Server{}
```

Comment:
Give that we say we are listening and just found the port. Is there a good reason not to start the server here? Doesn't starting the server here eliminate any chance of race conditions where the server isn't running yet but we have triggered the callback to the server?

Reply:
I kept the `Serve` in the original location because it is called after all the handlers are registered on the `mux` handler: `/login` and `callbackPath`. Then immediately after `Serve`, `OpenBrowser` logic occurs. If we move it to here before the handlers are registered, then there is a race as the handlers haven't been registered yet.